pkg/agent/agent_windows.go

```diff
@@ -111,6 +117,104 @@ func (i *Initializer) prepareHNSNetworkAndOVSExtension() error {
 	return util.PrepareHNSNetwork(subnetCIDR, i.nodeConfig.NodeTransportIPv4Addr, adapter, i.nodeConfig.UplinkNetConfig.Gateway, dnsServers, i.nodeConfig.UplinkNetConfig.Routes, i.ovsBridge)
 }
 
+func (i *Initializer) prepareVmNetworkAndOVSExtension() error {
+	var eNodeIface *net.Interface
+	en, err := i.crdClient.CrdV1alpha1().ExternalNodes(i.namespace).Get(context.TODO(), i.GetNodeConfig().Name, metav1.GetOptions{})
+	if err != nil {
+		klog.InfoS("Failed to get ExternalNode", "Name", i.GetNodeConfig().Name)
+		return err
+	}
+	for _, intF := range en.Spec.Interfaces {
+		//TODO: Handle multiple interface case
+		for _, addr := range intF.IPs {
+			var ipFilter *ip.DualStackIPs
+			epIP := net.ParseIP(addr)
+			if epIP.To4() != nil {
+				ipFilter = &ip.DualStackIPs{IPv4: epIP}
+			} else {
+				ipFilter = &ip.DualStackIPs{IPv6: epIP}
+			}
+			_, _, eNodeIface, err = util.GetIPNetDeviceFromIP(ipFilter, sets.NewString())
+			if err != nil {
+				klog.InfoS("Failed to get net device", "IP", addr)
+			} else {
+				klog.V(2).InfoS("Using the ExternalNode", "InterfaceName", eNodeIface.Name)
+				break
+			}
+		}
+		break
+	}
+
+	klog.V(2).InfoS("Setting up VM network")
+	// Check whether VM Switch is created
+	exists, err := util.VMSwitchExists()
+	if err != nil {
+		return err
+	}
+	if exists {
+		vmSwitchIfName, err := util.GetVMSwitchInterfaceName()
+		if err != nil {
+			return err
+		} else {
+			klog.InfoS("VM switch teaming", "interfaceName", vmSwitchIfName)
+			vmSwitchIfName = strings.TrimPrefix(vmSwitchIfName, util.PrefixPhy)
+			if eNodeIface.Name != vmSwitchIfName {
+				return fmt.Errorf("unexpected teaming interface %s found", vmSwitchIfName)
+			}
+			// Cache the uplink configuration
+			i.nodeConfig.UplinkNetConfig.Name = eNodeIface.Name
+		}
+		return nil
+	}
+
+	// upLinkIfName empty implies there is no VM switch
+	upLinkIfName := eNodeIface.Name
+	// Get the uplink interface configuration
+	upLinkIfConfig, err := util.GetUplinkConfig(upLinkIfName)
+	if err != nil {
+		return err
+	}
+	// Cache the uplink configuration
+	i.nodeConfig.UplinkNetConfig.Name = upLinkIfConfig.Name
+
+	var success = false
+	hostIfName := i.nodeConfig.UplinkNetConfig.Name
+	klog.InfoS("Using the interface", "hostIfName", hostIfName)
+	upLinkIfName = util.GenUplinkInterfaceName(hostIfName)
+	// Rename interfaceName to phy-interfaceName
```

Comment:
Now we should use a format "interfaceName~" to rename uplink.

Reply:
Done